specification/resources/resource-manager/Microsoft.Solutions/stable/2019-07-01/managedapplications.json

```diff
@@ -1606,16 +1606,39 @@
           "type": "string",
           "description": "The identity type.",
           "enum": [
-            "SystemAssigned"
+            "SystemAssigned",
+            "UserAssigned",
+            "SystemAssigned, UserAssigned",
+            "None"
           ],
           "x-ms-enum": {
             "name": "ResourceIdentityType",
             "modelAsString": false
           }
+        },
+        "userAssignedIdentities": {
+          "type": "object",
+          "additionalProperties": {
+            "$ref": "#/definitions/userAssignedResourceIdentity"
+          },
+          "description": "The list of user identities associated with the resource. The user identity dictionary key references will be resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}'."
         }
       },
       "description": "Identity for the resource."
     },
+    "userAssignedResourceIdentity": {
+      "properties": {
+        "principalId": {
+          "type": "string",
+          "description": "The principal id of user assigned identity."
+        },
+        "tenantId": {
```

Comment:
The public docs and the MSI spec seem to imply that the contract has changed to "principalId" and "clientId"... However, the contracts in ARM still have these properties.  Which is correct?  https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/Doc.aspx?sourcedoc=%7B66505FCE-8057-4524-B84F-4CEFE2D8A77C%7D&file=ARMResourcesWithStrongIdentity.docx&action=default&mobileredirect=true

I've seen more RPs using clientId recently

Reply:
This is inline with current service behavior, we are trying to make them sync for now.
I talked to the code owner of the the backend service, will submit a new api version and swagger when he makes the change.

Reply:
ok, please mark them as readOnly since the user is not providing these values